simulator/scheduler/scheduler.go

```diff
@@ -225,20 +238,12 @@ func convertConfigurationForSimulator(versioned *configv1.KubeSchedulerConfigura
 		versioned.Profiles[i].PluginConfig = pluginConfigForSimulatorPlugins
 	}
 
-	// Override the Extenders config so that the connection is directed to the simulator server.
-	extender.OverrideExtendersCfgToSimulator(versioned, simulatorPort)
-
-	defaultCfg, err := simulatorschedconfig.DefaultSchedulerConfig()
-	if err != nil {
-		return nil, xerrors.Errorf("get default scheduler config: %w", err)
-	}
+	apiconfigv1.SetDefaults_KubeSchedulerConfiguration(versioned)
 
-	// set default value to all field other than Profiles and Extenders.
-	defaultCfg.Profiles = versioned.Profiles
-	defaultCfg.Extenders = versioned.Extenders
-	versioned = defaultCfg
+	return versioned, nil
```

Comment:
These logic are moved to other funcs:

- convert from configv1 to config
- add default value
- filter out non-allowed changes